### ISSUE TITLE
sql: make all processor cores vectorize-able

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#57268]
Currently, we have native support for or can wrap many processor core types; however, there are some exceptions that prohibit the vectorization of the whole flow if the "bad" core is present in the physical plan. This issue tracks addressing all of those.
- [x] `LocalPlanNode`
   - there is a WIP to address it, #55909
- [x] `MetadataTestSender` and `MetadataTestReceiver`
   - I think we should remove these test processors entirely. Possibly #55758 is related.
- [ ] `ChangeAggregator` and `ChangeFrontier`
   - [blocked by](https://github.com/cockroachdb/cockroach/issues/55408#issuecomment-712350540).
- [x] `InvertedFilterer`
   - wrapping of the inverted filterer is unsupported because the cFetcher always decodes the data coming from disk, and the row-execution framework just happens to work due to `EncDatum.encoded` field. #50695 was the original issue filed for it, it is now closed, yet I think the problem is still present.
- [ ] `Backfiller`, `CSVWriter`, `Sampler`, `SampleAggregator`
   - these processors don't implement `execinfra.RowSource` interface, so they cannot be wrapped. They, however, use the DistSQL framework but are never - I think - planned in the flows containing other supported processors. Stats operators work is tracked in #41203.

Jira issue: CRDB-2842

**Comment by pbardea** [comment]
> these processors don't implement execinfra.RowSource interface, so they cannot be wrapped.

We have worked scheduled this release to migrate a lot of the "bulk processors" to use ProcessorBase and therefore implement `execinfra.RowSource`. (See https://github.com/cockroachdb/cockroach/issues/57293)

> however, use the DistSQL framework but are never - I think - planned in the flows containing other supported processors

This is indeed the case.

**Comment by yuzefovich** [comment]
I don't think we want to do this anymore, so closing as won't fix.
